### DISCUSSION TITLE
Regra de negócio no command e na entidade - Duplicação?

### DISCUSSION THREAD

**Discussion by WandersonOSantos** [balta-io/forum#20] [Tira Dúvidas] [answered]
Bom dia galera!!!!

Bom dia @andrebaltieri tudo beleza? 
Antes de mais nada queria deixar aqui meu agradecimento por todo conhecimento passado, pude evoluir bastante com os seus cursos.

Questão:

No fail fast validation temos a validação mínima de regras antes de executar qualquer ação no banco de dados ou outros processamentos visando poupar recursos.
Exemplo: Validar se um campo de descrição possui mais que quatro caracteres e menos que cinquenta, ou se um campo foi enviado com valores nulos.

A dúvida é:
Teremos essa mesma validação no command e na entidade? 
Se sim, não prejudicaria a manutenção? 
Seria errado deixar essa validação só no command?

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @WandersonOSantos, como vai?

Teremos essa mesma validação no command e na entidade?
- Depende da validação, pode acontecer sim, se for o caso você pode colocar esta validação em outra classe e chamar ela tanto na entidade quanto no Command.
-
Se sim, não prejudicaria a manutenção?
- Desde que não duplique o código, não veria problemas. Você pode inclusive extender o Flunt para ter mais validações se for o caso.

Seria errado deixar essa validação só no command?
- De forma alguma... Hoje no projeto do balta eu tenho diversas validações a nível de command apenas. O que faço neste caso é garantir que NADA entra sem ser via command e NADA sai sem ser via response.
